### PR TITLE
Buttons separators in RWD mobile should not be full height

### LINKED ISSUE #7704: Buttons separators in RWD mobile should not be full height
## 📝 Provide a description of the improvement

[We changed the way separators look in the toolbars](https://github.com/ckeditor/ckeditor5/issues/7493) to better integrate with balloons but we forgot about RWD forms (e.g. link):

These styles should also be updated if possible.

cc @panr 

---

If you'd like to see this improvement implemented, add a 👍 reaction to this post.

### PR DESCRIPTION
**Pull request by pkwasnik**
### Suggested merge commit message ([convention](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/contributing/git-commit-message-convention.html))

Fix (link): Implementing not full height separators. Closes #7704.

---

### Additional information

_For example – encountered issues, assumptions you had to make, other affected tickets, etc._

### CODE REVIEW

**Review by jodator — CHANGES_REQUESTED** [review]
I'm not sure about the approach, so I would double-check this with @oleq .

Apart from that :+1:  for picking up tests in the right way for the added element.

File: packages/ckeditor5-link/src/ui/linkformview.js [hunk 1]
```diff
@@ -24,6 +24,7 @@ import KeystrokeHandler from '@ckeditor/ckeditor5-utils/src/keystrokehandler';
 import checkIcon from '@ckeditor/ckeditor5-core/theme/icons/check.svg';
 import cancelIcon from '@ckeditor/ckeditor5-core/theme/icons/cancel.svg';
 import '../../theme/linkform.css';
+import ToolbarSeparatorView from '@ckeditor/ckeditor5-ui/src/toolbar/toolbarseparatorview';
```
[the remark this answers is not in the document]
Comment (pkwasnik) [reply packages/ckeditor5-link/src/ui/linkformview.js]: right, corrected

**Comment by oleq** [comment]
Aside from the fact that the solution breaks in certain situations



I think this direction is a dead-end. Some more thoughts:

1.  The action buttons are not a toolbar so using `ToolbarSeparatorView` in this context is a no-go.
2.  We cannot convert these buttons into a toolbar because... they are not a toolbar. Toolbars carry specific semantics and this is not the case.
3.  I'd see what we can do using `::before` or `::after` pseudo-selectors for the buttons (pure CSS approach). For instance `.ck-button::after`:

```
border-right: 1px solid var(--ck-color-base-border);
content: "";
width: 0;
position: absolute;
right: -1px;
top: var(--ck-spacing-small);
bottom: var(--ck-spacing-small);
z-index: 1;
```

results in something like this (ignore the broken toggle switches, it's just a PoC and I didn't narrow the selector)

**Review by oleq — CHANGES_REQUESTED** [review]
Aside from the fact that the solution breaks in certain situations



I think this direction is a dead-end. Some more thoughts:

1.  The action buttons are not a toolbar so using `ToolbarSeparatorView` in this context is a no-go.
2.  We cannot convert these buttons into a toolbar because... they are not a toolbar. Toolbars carry specific semantics and this is not the case.
3.  I'd see what we can do using `::before` or `::after` pseudo-selectors for the buttons (pure CSS approach). For instance `.ck-button::after`:

```
border-right: 1px solid var(--ck-color-base-border);
content: "";
width: 0;
position: absolute;
right: -1px;
top: var(--ck-spacing-small);
bottom: var(--ck-spacing-small);
z-index: 1;
```

results in something like this (ignore the broken toggle switches, it's just a PoC and I didn't narrow the selector)

**Review by jodator — CHANGES_REQUESTED** [review]
The mobile views are now OK :+1: 

However, the separator is now lost in manual decorators samples

PR | master
---|---
Selection_074 | Selection_075

@oleq - Previously we had this style copied 3 times (I see 3 deletions in this PR). For a single line style, it doesn't look like a big overhead. Adding more properties means that we have to copy them in 3 places. Do we have any strategy or pattern for duplicated definitions?

**Comment by oleq** [comment]
> Adding more properties means that we have to copy them in 3 places. Do we have any strategy or pattern for duplicated definitions?

Yeah. Normally we create a class and then it's shared across the use-cases. The story here is tricky, though because there's also the text alternative form that needs to be addressed and that means the styles must be shared across packages and cannot live in ckeditor5-link or ckeditor5-image. 



Some cake recipe (not sure it tastes good, though): 

*   Let's have `.ck-responsive-form` together with `.ck-text-alternative-form`, `.ck-link-form`, and `.ck-link-actions` (in `TextAlternativeFormView`, `LinkFormView`, `LinkActionsView`),
*   Let's have `ckeditor5-ui/theme/components/responsive-form/responsiveform.css` (for wireframe part), 
*   Let's have `ckeditor5-theme-lark/theme/ckeditor5-ui/components/responsive-form/responsiveform.css`  (for theme part). ATM both wireframe and theme CSS are in the same file. I really need to address #7702 to make it clear, sorry for that :disappointed: 
*   Let's make the `&::after { ... }` solution work as `.ck-responsive-form some-sane-selector-that-reaches-the-correct-button::after { ... }` both in responsive mode and when `.ck-link-form_layout-vertical` is used. This is tricky and needs to be thought through.
    *   It could be we need something like this:

```css
/* Either of responsiveform.css */

/* This one can be used here https://github.com/ckeditor/ckeditor5/blob/0706215c592edcf1b701b4af63298f4b77f8dd01/packages/ckeditor5-link/src/ui/linkformview.js#L139 */
/* It simply enforces the mode when the class is set. */
.ck-vertical-form some-sane-selector-that-reaches-the-correct-button::after {
	/* Some CSS. */
}

/* This one is smart and works based on media queries. */
@mixin ck-media-phone {
	.ck-responsive-form some-sane-selector-that-reaches-the-correct-button::after {
		/* Same CSS as above. */
	}
}
```

*   It all should work as an equivalent to code decorator pattern except that it's just CSS. There will be no JS (e.g. `ResponsiveFormView`) for the component, just CSS injected (first imported) via  `.ck-responsive-form` class.
*   We should make all the RWD/vertical styles common for these 3 forms in `responsiveform.css`, for instance, https://github.com/ckeditor/ckeditor5/blob/0706215c592edcf1b701b4af63298f4b77f8dd01/packages/ckeditor5-theme-lark/theme/ckeditor5-link/linkform.css#L29-L66. 
    *   I struggle to picture this all in my head without creating an actual code, though. But I think you got the idea.

**Comment by jodator** [comment]
> I really need to address #7702 to make it clear, sorry for that

It would be lovely if you could manage to find time for that :heart\_eyes:  having those written down would also mean fewer pings from others :wink:

File: packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/responsive-form/responsiveform.css [hunk 2]
```diff
@@ -1,16 +1,71 @@
...
+				width: 100%;
+			}
+
+			/* Let the long error messages wrap in the narrow form. */
+			& .ck-labeled-field-view__error {
+				white-space: normal;
+			}
+		}
+
+		& button.ck-button {
```
Comment (jodator) [inline packages/ckeditor5-theme-lark/theme/ckeditor5-ui/components/responsive-form/responsiveform.css:51]: @pkwasnik Could you add a comment on why there's `button.ck-button` is needed. It wasn't obvious to me why this is needed.

**Review by jodator — APPROVED** [review]
LGTM :+1: Only one request to @pkwasnik about documenting the `button.ck-button` usage and we can merge it.

**Review by jodator — APPROVED** [review]
LGTM :+1:

**Pull request merged**